src/nvim/screen.c

```diff
@@ -2196,7 +2198,7 @@ static int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool noc
 
     has_decor = decor_redraw_line(buf, lnum - 1, &decor_state);
 
-    providers_invoke_line(wp, providers, lnum - 1, &has_decor, &provider_err);
+    providers_invoke_line(wp, providers, lnum - 1, &has_decor, &decor_spell, &provider_err);
```

Comment:
I would have thought `decor_spell` would be set from `on_win`.

I think the purpose of the spell option in a decoration provider is just to turn off the legacy spell highlighting for that window. I don't think we should use it to control whether extmarks spell or not, they should always spell.

Reply:
> What about people using tree-sitter highlighting but don't use spell in general (like me)? Wouldn't they want to turn this off, too?

`set spell`

Reply:
> "This decoration provider handles spell checking on it's own"

Firstly, I don't see a reason to be this strict. Remember the only reason we are even adding this is to avoid the nasty hack.

Secondly, what prevents multiple decor providers from setting this option?

Right now all the option does is preventing extmarks set from outside the decor provider from setting spell marks.

Reply:
@clason : for context, the main reason we even need this option is to avoid this nasty hack: https://github.com/lewis6991/neovim/blob/45c20f32e24a9ed26093e9600402562b9a91377e/runtime/lua/vim/treesitter/spell.lua#L121-L128